compiler/rustc_passes/src/liveness.rs

```diff
@@ -1521,6 +1521,14 @@ impl<'tcx> Liveness<'_, 'tcx> {
     }
 
     fn warn_about_unused_args(&self, body: &hir::Body<'_>, entry_ln: LiveNode) {
+        if let Some(intrinsic) =
+            self.ir.tcx.intrinsic(self.ir.tcx.hir().body_owner_def_id(body.id()))
+        {
+            if intrinsic.must_be_overridden {
+                return;
+            }
+        }
```

Comment:
This seems to also contain the fix for https://github.com/rust-lang/rust/issues/135598? It'd be better to keep that separate.